File: src/main/java/it/giaquinto/springberry/controller/SpringBerryPi4JBaseController.java
package it.giaquinto.springberry.controller;

import it.giaquinto.springberry.component.Pi4JComponent;
import org.springframework.beans.factory.annotation.Autowired;

public abstract class SpringBerryPi4JBaseController extends SpringBerryBaseController {
    private Pi4JComponent pi4JComponent;

    @Autowired
    public void setPi4j(Pi4JComponent pi4JComponent) {
        this.pi4JComponent = pi4JComponent;
    }

    public Pi4JComponent getPi4j() {
        return pi4JComponent;
    }

}

File: src/main/java/it/giaquinto/springberry/controller/SpringBerryPi4JBaseController.kt
package it.giaquinto.springberry.controller

import it.giaquinto.springberry.component.Pi4JComponent
import org.springframework.beans.factory.annotation.Autowired

abstract class SpringBerryPi4JBaseController : SpringBerryBaseController() {
    @set:Autowired
    lateinit var pi4j: Pi4JComponent
}